electionleaflets/storages.py

```diff
@@ -34,3 +38,43 @@ def stored_name(self, name):
         if "?#" in name and not unparsed_name[3]:
             unparsed_name[2] += "?"
         return urlunsplit(unparsed_name)
+
+
+class TempUploadBaseMixin(abc.ABC):
+    """
+    This is a placeholder class that does nothing,
+    but other media classes will inherit from.
+
+    The point of this class is to give us something to assert against:
+    Leaflet uploads require a sub-class of this class to be in use, so we can
+    raise errors if someone attempts to use a storage class that doesn't
+    implement `save_from_temp_upload`.
+
+    """
+
+    @abc.abstractmethod
+    def save_from_temp_upload(self, source_path, target_file_path):
+        raise NotImplementedError
```

Comment:
This is a really minor point, but you don't need to `raise NotImplemented` in an abstract method. You can just `pass`

```py
import abc

class Parent(abc.ABC):
    @abc.abstractmethod
    def save_from_temp_upload():
        pass

class Child(Parent):
    pass

c = Child()
```

will throw

```
TypeError: Can't instantiate abstract class Child without an implementation for abstract method 'save_from_temp_upload'
```